tests/roles/ovn_adoption/tasks/main.yaml

```diff
@@ -29,31 +29,77 @@
   retries: 60
   delay: 2
 
-- name: get podified OVN NB ovsdb-server IP
+- name: get podified OVN NB ovsdb-server service cluster IP
   ansible.builtin.shell: |
     {{ shell_header }}
     {{ oc_header }}
-    oc get pod ovsdbserver-nb-0 -o jsonpath='{.metadata.annotations.k8s\.v1\.cni\.cncf\.io/network-status}' | \
-    jq 'map(. | select(.name=="openstack/internalapi"))[0].ips[0]' | \
-    tr -d '"'
+    oc get svc --selector "statefulset.kubernetes.io/pod-name=ovsdbserver-nb-0" -ojsonpath='{.items[0].spec.clusterIP}'
   register: podified_ovn_nb_ip_result
 
 - name: get podified OVN SB ovsdb-server IP
   ansible.builtin.shell: |
     {{ shell_header }}
     {{ oc_header }}
-    oc get pod ovsdbserver-sb-0 -o jsonpath='{.metadata.annotations.k8s\.v1\.cni\.cncf\.io/network-status}' | \
-    jq 'map(. | select(.name=="openstack/internalapi"))[0].ips[0]' | \
-    tr -d '"'
+    oc get svc --selector "statefulset.kubernetes.io/pod-name=ovsdbserver-sb-0" -ojsonpath='{.items[0].spec.clusterIP}'
   register: podified_ovn_sb_ip_result
 
+- name: set fact for oc run args
+  ansible.builtin.set_fact:
+    oc_run_args:
+      spec:
+        containers:
+        - image: "{{ ovn_image }}"
+          command: [ "sh", "-c", "sleep infinity"]
+          name: adoption
+          volumeMounts:
+          - mountPath: /backup
+            name: backup
+        securityContext:
+          allowPrivilegeEscalation: false
+          capabilities:
+            drop: ALL
+          runAsNonRoot: true
+          runAsUser: 1000
+          seccompProfile:
+            type: RuntimeDefault
+        volumes:
+        - hostPath:
+            path: "{{ playbook_dir }}/{{ ovn_copy_tmp_dir }}"
+            type: Directory
+          name: backup
+
+- name: prepare oc run args file
+  copy:
+    content: "{{ oc_run_args }}"
+    dest: "{{ ovn_copy_tmp_dir }}/oc_run_args.yaml"
```

Comment:
the `ovn_copy_tmp_dir` is not created until the `dump OVN databases` task later on, we need to move the creation before this task